.github/workflows/docs.yml

```diff
@@ -1,36 +1,57 @@
-name: Documentation
+name: Docs CI
 
-on: [push, pull_request]
+on:
+  push:
+  pull_request:
 
 jobs:
-  build:
-
+  docs:
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [3.8]
-      fail-fast: false
 
     steps:
-
-    - uses: actions/checkout@v2
-
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
-
-    - name: Install
-      shell: bash -l {0}
-      run: source continuous_integration/scripts/install.sh
-
-    - name: Install documentation-building requirements
-      shell: bash -l {0}
-      run: |
-        set -vxeuo pipefail
-        python -m pip install -r requirements-dev.txt
-        python -m pip list
-
-    - name: Build Docs
-      shell: bash -l {0}
-      run: make -C docs/ html
+      - name: Avoid git conflicts when tag and branch pushed at same time
+        if: startsWith(github.ref, 'refs/tags')
+        run: sleep 60
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Need this to get version number from last tag
+          fetch-depth: 0
+
+      - name: Install system packages
+        # Can delete this if you don't use graphviz in your docs
+        run: sudo apt-get install graphviz
+
+      - name: Install python packages
+        uses: ./.github/actions/install_requirements
+        with:
+          requirements_file: requirements-dev-3.x.txt
+          install_options: -e .[dev]
+
+      - name: Build docs
+        run: tox -e docs
+
+      - name: Sanitize ref name for docs version
+        run: echo "DOCS_VERSION=${GITHUB_REF_NAME//[^A-Za-z0-9._-]/_}" >> $GITHUB_ENV
+
+      - name: Move to versioned directory
+        run: mv build/html .github/pages/$DOCS_VERSION
+
+      - name: Write switcher.json
+        run: python .github/pages/make_switcher.py --add $DOCS_VERSION ${{ github.repository }} .github/pages/switcher.json
+
+      - name: Deploy documentation to blueskyproject.io.
+        if: github.event_name == 'push' && github.actor != 'dependabot[bot]'
+        # We pin to the SHA, not the tag, for security reasons.
+        # https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions
+        uses: peaceiris/actions-gh-pages@bbdfb200618d235585ad98e965f4aafc39b4c501  # v3.7.3
+        with:
+          deploy_key: ${{ secrets.ACTIONS_DOCUMENTATION_DEPLOY_KEY }}
+          publish_branch: master
+          publish_dir: ./docs/build/html
+          external_repository: bluesky/bluesky.github.io
+          destination_dir: ${{ env.REPOSITORY_NAME }}  # just the repo name, without the "bluesky/"
+          keep_files: true
```

Comment:
Please can you put this back to the original deply stage, so it looks like https://github.com/bluesky/ophyd-epics-devices/blob/3f87aa5f09a94eebcae1ed132fc295b690a59c50/.github/workflows/docs.yml#L45-L53

This is because we can actually publish to the local gh-pages branch, and if it doesn't exist in bluesky/bluesky.github.io then it will redirect to the gh-pages of this repo

Reply:
I'm a little confused at this: [this is the original deploy](https://github.com/bluesky/event-model/blob/a89364fd53fae0a80599bac7cca3132e4c5176de/.github/workflows/publish-docs.yml#L49)